apps/web/src/utils/nodeToString.ts

```diff
@@ -0,0 +1,24 @@
+import { createRoot } from "react-dom/client";
+import { flushSync } from "react-dom";
+import { ReactNode } from "react";
+
+/**
+ * Convert a ReactNode to a string
+ *
+ * REF: https://react.dev/reference/react-dom/server/renderToString#removing-rendertostring-from-the-client-code
+ */
+function nodeToString(node: ReactNode): string {
+  if (typeof node === "string") return node;
+
+  const div = document.createElement("div");
+  const root = createRoot(div);
+  queueMicrotask(() => {
```

Comment:
I've never used queueMicrotask before.  It looks like this is your own enhancement to the function?  How does this work?

Reply:
So, I was getting an error that `nodeToString` was being called in a life cycle hook (you cannot call `flushSync` in a  life cycle hook to [avoid the many caveats](https://react.dev/reference/react-dom/flushSync#caveats))

The `queueMicroTask` basically ripped this out of the life cycle by queuing it after it occurred. However, I can not reproduce the error so I removed it in [remove micro task](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11662/commits/5e735ca5b15f22e2b87189f32b569c9afd30d6bd)